### PR TITLE
Add Positive test cases for Content Views via API - Part 3

### LINKED ISSUE #1266: Add Positive Create test cases for Content Views via API - Part 3
Using the new API layer, please create `data-driven` (mostly for `name` attribute) for *Content Views* using the following guidelines for test names, steps and assertions:

The following tests should create a `Content View` with *ONE* custom (*Puppet*) product and *ONE* puppet module

* Publish it once
  * Check that it was published
  * Check that it has only ONE version
  * Check that it has Puppet modules
* Publish it (random > 1) times
  * Check that it was published as many times
  * Check that it has N versions
  * Check that each version has Puppet modules
* Publish it and promote it once
  * Check that it was published
  * Check that it has only ONE version
  * Check that it was promoted to Library + 1 environment
  * Check that environment has Puppet modules
* Publish it and promote it to Library + (random > 1) environments
  * Check that it was published as many times
  * Check that it has ONE version
  * Check that it was promoted to random environment
  * Check that random environment has Puppet modules

### PR DESCRIPTION
**Pull request by oshtaier**
Added API test case to Resolve #1266

### CODE REVIEW

File: tests/foreman/api/test_contentview.py [hunk 1]
```diff
@@ -5,11 +5,15 @@
...
-from robottelo.common.constants import PUPPET_MODULE_NTP_PUPPETLABS
+from robottelo.common.constants import (
+    FAKE_0_PUPPET_REPO,
+    PUPPET_MODULE_NTP_PUPPETLABS
+)
 from robottelo.common.decorators import (
     bz_bug_is_open, data, run_only_on, stubbed)
 from robottelo.common.helpers import get_data_file, get_server_credentials
 from robottelo.test import APITestCase
+import random
```
Comment (elyezer) [inline tests/foreman/api/test_contentview.py]: Please, move import statements before any `from ... import`. They should look like the following:

```python
import a
import b
import c

from d import x
from e import y
```

File: tests/foreman/api/test_contentview.py [hunk 2]
```diff
@@ -287,19 +293,43 @@ def test_positive_publish_2(self):
             cvv = entities.ContentViewVersion(id=cvv_id)
             self.assertGreater(cvv.read_json()['package_count'], 0)
 
-    def test_positive_publish_3(self):
-        """@Test: Publish a content view that has puppet modules several times.
+    def test_publish_cv_with_puppet_once(self):
```
Comment (elyezer) [inline tests/foreman/api/test_contentview.py]: thanks for updating the name

**Comment by elyezer** [comment]
Looks good, ACK.

File: tests/foreman/api/test_contentview.py [hunk 3]
```diff
@@ -310,8 +341,9 @@ def test_positive_publish_3(self):
 
         # Publish the content view several times and check that each version
         # has the puppet module added above.
-        for _ in range(REPEAT):
+        for i in range(random.randint(3, 5)):
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentview.py]: I very much like random data. However, publishing and promoting content views takes an enormously long time. Here's the comment for `REPEAT` (it should be turned in to a docstring):

    # Some tests repeatedly publish content views or promote content view versions.
    # How many times should that be done? A higher number means a more interesting
    # but longer test.

Is there a compelling reason to use `random.randint` here?

Comment (Ichimonji10) [inline tests/foreman/api/test_contentview.py]: Ahh. I've read through #1266 and see why this change has been made. Never mind.

**Comment by Ichimonji10** [comment]
ACK
